### ISSUE TITLE
Come up with clear and consistent naming for "currently selected" vs "confirmed picked" items

### ISSUE THREAD

**Issue by crumblingstatue** [fluxxcode/egui-file-dialog#171]
#170 Adds methods to get the currently selected directory entries, but the naming of the methods is not ideal, and might be a source of confusion for users.

File dialogs are often referred to as file pickers, so perhaps what is right now referred to as "selection" (`select_file`, `selected`, `take_selected`, ...) could be renamed to `pick_file`, `picked`, `take_picked`, etc.
The word "pick" implies that the user has already made up their mind, as opposed to just select, which can be an action during the picking process (selecting multiple files to pick, etc.).

There might be other good candidates as well. But after careful consideration, we should pick a consistent naming scheme, and then review all the relevant API items and rename everything where appropriate. This will be a breaking change of course, so it should be done sooner rather than later.

**Comment by fluxxcode** [comment]
I completely agree. I'm currently wondering whether it would be better to put this into a separate release without any further changes.
I think I'll change it with `v0.9.0`.

**Comment by fluxxcode** [comment]
Roadmap:
`v0.8.0`:
- [ ] Add new `pick_directory`, `pick_file`, `pick_multiple`, `picked`, `take_picked`, `picked_multiple`, `take_picked_multiple`
- [ ] Deprecate `select_directory`, `select_file`, `select_multiple`, `selected`, `take_selected`, `selected_multiple`, `take_selected_multiple`

`v0.9.0`:
- [ ] Remove deprecated methods
- [ ] Rename `DialogMode`'s: `SelectFile` -> `PickFile`, `SelectDirectory` -> `PickDirectory`, `SelectMultiple` -> `PickMultiple`
